benchadapt/python/benchadapt/adapters/archery.py

```diff
@@ -9,9 +12,12 @@ class ArcheryAdapter(GoogleBenchmarkAdapter):
     """A class for running Apache Arrow's archery benchmarks and sending the results to conbench"""
 
     def __init__(self) -> None:
-        super().__init__(command=["archery", "benchmark", "run"])
+        result_file = NamedTemporaryFile()
```

Comment:
ah shoot you know what? I think this should be an attribute too (could be hidden). Or else it could get garbage collected after exiting `__init__()` and subsequent writes to the file might fail. 

Unless you're planning never to write to this file if the benchmarks are supposed to override it